ibm/service/database/resource_ibm_database.go

```diff
@@ -1565,57 +1565,38 @@ func resourceIBMDatabaseInstanceCreate(context context.Context, d *schema.Resour
 		}
 	}
 
-	if wl, ok := d.GetOk("whitelist"); ok {
-		whitelist := flex.ExpandWhitelist(wl.(*schema.Set))
-		for _, wlEntry := range whitelist {
-			whitelistReq := icdv4.WhitelistReq{
-				WhitelistEntry: icdv4.WhitelistEntry{
-					Address:     wlEntry.Address,
-					Description: wlEntry.Description,
-				},
-			}
-			task, err := icdClient.Whitelists().CreateWhitelist(icdId, whitelistReq)
-			if err != nil {
-				return diag.FromErr(fmt.Errorf("[ERROR] Error updating database whitelist entry: %s", err))
-			}
-			_, err = waitForDatabaseTaskComplete(task.Id, d, meta, d.Timeout(schema.TimeoutCreate))
-			if err != nil {
-				return diag.FromErr(fmt.Errorf(
-					"[ERROR] Error waiting for update of database (%s) whitelist task to complete: %s", icdId, err))
-			}
+	_, hasWhitelist := d.GetOk("whitelist")
+	_, hasAllowlist := d.GetOk("allowlist")
+
+	if hasWhitelist || hasAllowlist {
+		var ipAddresses *schema.Set
+		if hasWhitelist {
+			ipAddresses = d.Get("whitelist").(*schema.Set)
+		} else {
+			ipAddresses = d.Get("allowlist").(*schema.Set)
 		}
```

Comment:
Since we are moving on to `allowlist` should that get preference here?
ie,

```
		if hasAllowlist {
			ipAddresses = d.Get("allowlist").(*schema.Set)
		} else {
			ipAddresses = d.Get("whitelist").(*schema.Set)
		}

```

Reply:
@obai-1 Only one will be set since these attributes conflict with each other

Reply:
Gotcha